quixstreams/sources/kafka/kafka.py

```diff
@@ -0,0 +1,332 @@
+import time
+import logging
+
+from typing import Union, Optional, Dict, TYPE_CHECKING
+
+from confluent_kafka import TopicPartition, Message
+
+from quixstreams.kafka import Consumer, ConnectionConfig, AutoOffsetReset
+from quixstreams.kafka.exceptions import KafkaConsumerException
+from quixstreams.error_callbacks import default_on_consumer_error, ConsumerErrorCallback
+from quixstreams.models.topics import TopicConfig, TopicAdmin, Topic
+from quixstreams.models.serializers import DeserializerType
+from quixstreams.sources import Source
+
+from .checkpoint import Checkpoint
+
+if TYPE_CHECKING:
+    from quixstreams.app import ApplicationConfig
+
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["KafkaSource"]
+
+
+class KafkaSource(Source):
+    """
+    Source implementation that replicate a topic from a kafka cluster to your application cluster.
```

Comment:
```suggestion
    A Source implementation that replicates a topic from a separate Kafka broker to your Application broker.
    
    It can copy messages for development and testing without risking producing them back or affecting the consumer groups.
```

Reply:
> It can copy messages for development and testing without risking producing them back or affecting the consumer groups.

This should should be fine for production use. I don't think this is necessary.